_posts/2018-09-13-is-cockpit-secure.md

```diff
@@ -0,0 +1,257 @@
+---
+title: Is Cockpit Secure?
+date: 2018-09-13 10:00
+category: blog
+tags: cockpit
+slug: is-cockpit-secure
+---
+
+Security is of your Linux systems vital. It's not a binary thing though.
+Depending on your requirements you end up choosing a level of security that
+still allows you and your systems to accomplish what they need to do.
+
+Here's info about [Cockpit's](http://cockpit-project.org/) security, to help
+you make those choices. You'll find not only has Cockpit got a solid security
+story, but you can use it in all sorts of different ways depending on what
+kind of security your systems need.
+
+## Cockpit is a Linux session in your browser
+
+Let start by stacking the deck against Cockpit. First off it's important to
+remember that Cockpit is actually an alternative Linux session ... along side X11,
+SSH and VT logins. It's a session in the [PAM](http://www.linux-pam.org/),
+[logind](https://www.freedesktop.org/wiki/Software/systemd/logind/), TTY,
+[SELinux](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Security-Enhanced_Linux/sect-Security-Enhanced_Linux-Confining_Users-Confining_Existing_Linux_Users_semanage_login.html)
+ and other senses.
+The logged in browser and the javascript code it is running can interact directly
+with the system as part of that session.
+
+Try this out if you want. [Log into Cockpit](http://cockpit-project.org/running.html),
+and pop up the Javascript console by pressing *Ctrl-Shift-J* in your browser. Now type
+commands like the following:
+
+    > zz = cockpit.spawn(["id"])
```

Comment:
This only returns a promise now, doesn't it? How about

```
zz = cockpit.spawn(["id"]).then(console.log)
```

Reply:
Assigning to zz does all that magic. Try it.